compiler/noirc_evaluator/src/ssa/opt/remove_unreachable_instructions.rs

```diff
@@ -0,0 +1,465 @@
+            jmpif u1 0 then: b4, else: b1
+          b4():
+            v1 = add Field 1, Field 2
+            return v1
+        }
+        "#;
+        let ssa = Ssa::from_str(src).unwrap();
+        let ssa = ssa.remove_unreachable_instructions();
+
+        assert_ssa_snapshot!(ssa, @r#"
+        acir(inline) predicate_pure fn main f0 {
+          b0():
+            jmp b1()
+          b1():
+            v2 = add Field 1, Field 2
+            jmp b2(v2)
+          b2():
+            jmpif u1 0 then: b3, else: b4
+          b3():
+            constrain u1 0 == u1 1, "Index out of bounds"
+            unreachable
+          b4():
+            v4 = add Field 1, Field 2
+            return v4
+        }
+        "#);
+    }
+
+    #[test]
+    fn does_not_zeroes_terminator_of_non_dominated_block_2() {
+        // Here b3 is a successof of b2 but is not dominated by it.
+        let src = r#"
+        acir(inline) predicate_pure fn main f0 {
+          b0():
+            jmp b1()
+          b1():
+            jmpif u1 0 then: b2, else: b3
+          b2():
+            constrain u1 0 == u1 1, "Index out of bounds"
+            jmp b3()
+          b3():
+            v1 = add Field 1, Field 2
+            return v1
+        }
+        "#;
+        let ssa = Ssa::from_str(src).unwrap();
+        let ssa = ssa.remove_unreachable_instructions();
+
+        assert_ssa_snapshot!(ssa, @r#"
+        acir(inline) predicate_pure fn main f0 {
+          b0():
+            jmp b1()
+          b1():
+            jmpif u1 0 then: b2, else: b3
+          b2():
+            constrain u1 0 == u1 1, "Index out of bounds"
+            unreachable
+          b3():
+            v3 = add Field 1, Field 2
+            return v3
+        }
+        "#);
+    }
+
+    #[test]
+    fn does_not_zeroes_terminator_of_non_dominated_block_3() {
+        // Here b4 is a transitive successor of b2 but is not dominated by it.
+        let src = r#"
+        acir(inline) predicate_pure fn main f0 {
+          b0():
+            jmp b1()
+          b1():
+            jmpif u1 0 then: b2, else: b3
+          b2():
+            constrain u1 0 == u1 1, "Index out of bounds"
+            jmp b4()
+          b3():
+            jmp b4()
+          b4():
+            v1 = add Field 1, Field 2
+            return v1
+        }
+        "#;
+        let ssa = Ssa::from_str(src).unwrap();
+        let ssa = ssa.remove_unreachable_instructions();
+
+        assert_ssa_snapshot!(ssa, @r#"
+        acir(inline) predicate_pure fn main f0 {
+          b0():
+            jmp b1()
+          b1():
+            jmpif u1 0 then: b2, else: b3
+          b2():
+            constrain u1 0 == u1 1, "Index out of bounds"
+            unreachable
+          b3():
+            jmp b4()
+          b4():
+            v3 = add Field 1, Field 2
+            return v3
+        }
+        "#);
+    }
+
+    #[test]
+    fn does_not_zeroes_terminator_of_non_dominated_block_4() {
```

Comment:
Looks like some tests names still mention the old approach for zeroing terminators